### PR TITLE
Bluetooth: controller: Fix race in create connection cancel

### PR DESCRIPTION
**Pull request by cvinayak**
Fix race conditional when create connection cancel is called
and actually a connection did get setup while initiator is
being stopped.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Comment by carlescufi** [comment]
@kruithofa @mtpr-ot  @wopu-ot @thoh-ot please review

**Review by kruithofa — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_conn.c [hunk 1]
```diff
@@ -800,6 +800,9 @@ void ull_conn_setup(memq_link_t *link, struct node_rx_hdr *rx)
 
 	ftr = &(rx->rx_ftr);
 
+	/* NOTE: LLL conn context SHALL be after lll_hdr in
+	 *       struct lll_adv and struct lll_scan.
+	 */
 	lll = *((struct lll_conn **)((uint8_t *)ftr->param +
```
Comment (mtpr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_conn.c:806]: Just a note for later discussion: Race with ADV disable may invalidate `ftr->param`, making it `NULL` by the time this is called. `lll->role` is then invalid.

**Review by mtpr-ot — APPROVED** [review]
Nice fix, cleanup and code commenting (comments much appreciated)

File: subsys/bluetooth/controller/ll_sw/ull_conn.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_conn.c:806]: Agreed, do you have a fix already that you could send a PR?

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_conn.c:806]: Create this to keep track: https://github.com/zephyrproject-rtos/zephyr/issues/33868

**Pull request merged**